source/adapters/level_zero/command_buffer.cpp

```diff
@@ -135,6 +160,35 @@ ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
   for (auto &ZeFence : ZeFencesList) {
     ZE_CALL_NOCHECK(zeFenceDestroy, (ZeFence));
   }
+
+  auto ReleaseIndirectMem = [](ur_kernel_handle_t Kernel) {
```

Comment:
Is this something we should have been doing before, or did a part of this PR introduce the need for it?

Reply:
Before kernels were attached to the event returned by the command enqueuing and were released when the event was cleaned up. As we do not have events when in-order command lists are used, we now have to manage the release of the kernels ourselves.